config/initializers/omniauth.rb

```diff
@@ -1,3 +1,48 @@
+module OmniAuth
+  module Strategies
+    class OpenIDConnect
+      # rubocop:disable Metrics/AbcSize
+      # rubocop:disable Metrics/CyclomaticComplexity
+      # rubocop:disable Metrics/PerceivedComplexity
+      # rubocop:disable Metrics/LineLength
+      # rubocop:disable Style/GuardClause
+      # Please refer to this commit to read why this has been copied from: https://github.com/m0n9oose/omniauth_openid_connect/blob/master/lib/omniauth/strategies/openid_connect.rb
+      def callback_phase
+        error = request.params['error_reason'] || request.params['error']
+        if error
+          raise CallbackError.new(request.params['error'], request.params['error_description'] || request.params['error_reason'], request.params['error_uri'])
+        elsif request.params.blank?
+          redirects = { '/auth/dfe/callback' => '/dfe/sessions/new' }
+          req = Rack::Request.new(env)
+          return redirect(redirects[req.path]) if redirects.include?(req.path)
+        elsif request.params['state'].to_s.empty? || request.params['state'] != stored_state
+          return redirect('/401')
+        elsif !request.params['code']
+          return fail!(:missing_code, OmniAuth::OpenIDConnect::MissingCodeError.new(request.params['error']))
+        else
+          options.issuer = issuer if options.issuer.blank?
```

Comment:
can we use `||=` instead of an if statement

Reply:
This code has been directly copy and pasted from the omniauth-connect gem so I was reluctant to rewrite any of it for style reasons.